url53f943c5cb_Praznat_AGP_tgz-pJ8-AMath_TestingJ8/src/GUI/TextDisplay/QuestScroll.java

```diff
@@ -0,0 +1,45 @@
+package GUI.TextDisplay;
+
+import java.awt.Graphics;
+
+import Descriptions.GobLog.Reportable;
+import GUI.PopupAbstract;
+import Questing.Quest;
+import Sentiens.Clan;
+
+public class QuestScroll extends Papyrus {
+	private static final int REPORTSTART = 10;
+	public QuestScroll(PopupAbstract P) {
+		super(P);
+	}
+	public void paint(Graphics g) {
+		Clan clan = clan();
+		if(clan==null){return;}
+		super.paint(g);
+		int r = 0;
+		String S;
+		for(Quest q : clan.MB.QuestStack) { //reverse order?
+			g.drawString(S = q.description(), 2, BHGT+BHGT*r++);
+			refreshWid(g, S);
+		}
+		r = REPORTSTART;
+		g.drawString("GobLog:", 2, BHGT+BHGT*r++);
+		Reportable[] report = clan.getLog();
+		int d = -1;
+		for (Reportable R : report) {
+			int rd = R.getDate();
+			if (rd != d) {d = rd; g.drawString("Day " + rd, 2, BHGT+BHGT*r++);}
+			S = R.toString();
+			if (S != "") {g.drawString(S, 2, BHGT+BHGT*r++);}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.